src/components/NewPost.js

```diff
@@ -1,13 +1,48 @@
+import { useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { createPost } from "../features/post/postSlice";
 const NewPost = () => {
+  const { authToken, user } = useSelector((state) => state.auth);
+  const { avatarURL, firstName, lastName, username } = user;
+  console.log(user)
+  const [userPost, setUserPost] = useState({
+    avatarURL,
+    firstName,
+    lastName,
+    username,
+    comments: [],
+    content: "",
+  });
+
+  function postHandler(postData, authToken) {
+    if (postData.content !== "" && postData.content.replace(/\s/g, "")) {
+      dispatch(createPost({ postData, authToken }));
+      
+    }
+    setUserPost({
+      avatarURL,
+      firstName,
+      lastName,
+      username,
+      comments: [],
+      content: "",
+    });
+  }
+  const dispatch = useDispatch();
   return (
     <div className="border-b border-gray-500 p-3 flex space-x-3 overflow-y-scroll scrollbar-hide">
       <img
-        src="https://images.pexels.com/photos/220453/pexels-photo-220453.jpeg?cs=srgb&dl=pexels-pixabay-220453.jpg&fm=jpg"
+        src={avatarURL}
```

Comment:
Can you add a default URL if user doesn't have any avatar set

Reply:
Thanks for pointing it out, I will add it.